gateway-service-metadata/src/main/java/org/apache/knox/gateway/service/metadata/KnoxMetadataResource.java

```diff
@@ -101,6 +103,7 @@ public GeneralProxyInformation getGeneralProxyInformation() {
       final ServerInfoService serviceInfoService = gatewayServices.getService(ServiceType.SERVER_INFO_SERVICE);
       final String versionInfo = serviceInfoService.getBuildVersion() + " (hash=" + serviceInfoService.getBuildHash() + ")";
       proxyInfo.setVersion(versionInfo);
+      proxyInfo.setHostname(Hostname.getHostname());
```

Comment:
Wouldn't a simple `InetAddress.getLocalHost().getHostName()` work? Do we know how is this library implemented?

Reply:
See explanation here: https://github.com/mattsheppard/gethostname4j

Reply:
We've been already using `InetAddress.getLocalHost().getHostName()` and `InetAddress.getLocalHost().getCanonicalHostName()` at a couple of places in the project. 

I wonder how critical is to have the real hostname in cases like this. If this is better we might need to consider replacing the existing references to these `InetAddress` calls. 

Otherwise if this is just a nice to have information, we might    need to think about using the existing idom, instead of introducing a new dependency.

Since adding a new dependency is always a liability, for future maintenance and possible CVE point of view.

Reply:
> I wonder how critical is to have the real hostname in cases like this.

In the case of troubleshooting in a real cluster, this can be critical.

I agree, we may revisit the current usage of `InetAddress.getLocalHost()` and replace them, if needed, with this new library.

WRT. the CVE concern, I also had that in my mind, but at the end of the day, this new dependency does nothing special but delegate the `gethostname` call to the OS-native layer using JNA. This is a 1-class dependency, I cannot see any reported vulnerabilities so far.
<img width="1220" alt="Screenshot 2024-02-27 at 9 37 05" src="https://github.com/apache/knox/assets/34065904/d4b08953-c677-400c-a77c-19361df82a6a">